src/include/catalog/index_schema.h

```diff
@@ -280,6 +503,30 @@ class IndexSchema {
     ExtractIndexedColOids();
   }
 
+  /**
+   * Instantiates a new catalog description of an index
+   * @param columns describing the individual parts of the key
+   * @param type backing data structure of the index
+   * @param is_unique indicating whether the same key can be (logically) visible repeats are allowed in the index
+   * @param is_primary indicating whether this will be the index for a primary key
+   * @param is_exclusion indicating whether this index is for exclusion constraints
+   * @param is_immediate indicating that the uniqueness check fails at insertion time
+   * @param index_options that are options for building the index
+   */
+  IndexSchema(std::vector<Column> columns, const storage::index::IndexType type, const bool is_unique,
```

Comment:
I'm a little bit hesitant about adding a new `IndexSchema` constructor. Can we remove the old constructor and then passing in an empty `index_options` when not applicable?